### ISSUE TITLE
Plugin crashes in Carla

### ISSUE THREAD

**Issue by yurivict** [sjaehn/BShapr#10]
Please see the stack here: https://github.com/falkTX/Carla/issues/973

**Comment by sjaehn** [comment]
Thanks for reporting and the helpful backtrace. It's in work now.

**Comment by sjaehn** [comment]
Does this 4ca7f675a5fb31f507c626a748dad731ab447dd2 help? I've to ask you as I didn't had crashes on my systems. But I reproduced your backtrace, especially the suspicious double destructor calls.
```
#3  0x00000008012698f4 in BWidgets::FocusWidget::~FocusWidget (this=0x8021a7c00) at src/BWidgets/FocusWidget.cpp:33
#4  0x0000000801269989 in BWidgets::FocusWidget::~FocusWidget (this=0x8021a7c00) at src/BWidgets/FocusWidget.cpp:33
```

If this helps, I've to do some additional cleanup. Or maybe better to throw away the whole FocusWidget concept (used for hover effect).

**Comment by yurivict** [comment]
You can also run it under Valgrind and/or Google Perftools to make sure memory operations aren't faulty.

**Comment by sjaehn** [comment]
Solved in 1c5ed6d9b353b5addc9a49339acb21b33f438398 , 63fa5da26e9212c6892086c4a80e9f45c87d8a79 , and finally 63fa5da26e9212c6892086c4a80e9f45c87d8a79 . The replacement of the badly hacked FocusWidgets required major changes in > 2k code lines in the BWidgets TK.
